File: zigar-runtime/test/doc-examples/print-struct-defaults.js
// print-struct-defaults.js
import { printStruct } from './print-struct-defaults.zig';

printStruct({ a: {}, number3: 77 });
printStruct({ number3: 77 });

// console output:
// print-struct-defaults.StructB{ .a = print-struct-defaults.StructA{ .number1 = 1, .number2 = 2 }, .number3 = 7.7e+01 }
// print-struct-defaults.StructB{ .a = null, .number3 = 7.7e+01 }

File: zigar-runtime/test/doc-examples/print-struct-defaults.zig
// print-struct-defaults.zig
const std = @import("std");

pub const StructA = struct {
    number1: i16 = 1,
    number2: i16 = 2,
};

pub const StructB = struct {
    a: ?StructA = null,
    number3: f64 = 3,
};

pub fn printStruct(s: StructB) void {
    std.debug.print("{any}\n", .{s});
}
